### ISSUE TITLE
Blocks: Sponsors not consistent with other blocks

### ISSUE THREAD

**Issue by iandunn** [WordPress/wordcamp.org#57]
- [x] `includes/sponsors.php` missing `add_script_data` and `get_options`
- [x] missing `toolbar.js`, `{type}-select.js`, `block-content.scss`
- [x] `block-content.js`, `block-controls` have lots of differences
- [x] `withSelect` vs `apiFetch` (being addressed in https://github.com/WordPress/wordcamp.org/pull/44)
- [ ] I haven't checked every line, so there may be some other things that are missing/inconsistent as well.
- [x] `Notice: Undefined index: className in mu-plugins/blocks/includes/sponsors.php on line 41` because missing the default arguments that other blocks have

What's the reason for those differences? 

If the way that Sponsors does thing is better, we should update the other blocks to do the same. In the cases where there isn't a compelling reason to do it differently, I think it'll be more important to maintain consistency, so Sponsors should be updated to adopt the conventions of the other blocks.

**Comment by iandunn** [comment]
I noticed other inconsistencies, so I updated the title and description to make this more generic.

**Comment by coreymckrill** [comment]
Bringing this over from [a discussion](https://github.com/WordPress/wordcamp.org/pull/33#pullrequestreview-216423357) on #33:

> If the block schema gets defined in PHP and loaded via add_script_data, ... it would be simple to use the schema values instead of a static variable. That would potentially make it DRYer as well.

This is in relation to hard-coding the schema for the Grid Layout component.

**Comment by vedanshujain** [comment]
> `includes/sponsors.php` missing `add_script_data` and `get_options`

Sponsor block should be updated to add this, and fetch this default props data from backend. Further, props data from all shared component should be defined at only one place in backend, duplicating it everywhere does not makes sense. 

> missing `toolbar.js`, `{type}-select.js`,`block-content.scss`

`toolbar.js` is not needed because toolbar was adding grid settings, which is now provided by `grid-layout/toolbar.js`. We can go ahead and remove this from blocks who do not have any toolbar setting other than layout.

`{type}-select` Sponsor does not have it because I was thinking most of the stuff can be abstracted out by `ItemSelect` so we may won;t need it. But it does not look like thats the case, so lets refactor Sponsor's `block-controls.js` to make it consistent with other blocks. 

`block-content.scss` is not needed for Sponsors block because it does not need any special styling.

> `block-content.js`, `block-controls` have lots of differences

This mostly because of `withSelect` Its being tracked in #65 .

**Comment by iandunn** [comment]
I added another item to the checklist that I ran into today with `$attributes['className']` not having a default argument.

**Comment by coreymckrill** [comment]
> $attributes['className'] not having a default argument

I think this is because the PHP function `get_attributes_schema` doesn't have a definition for `className`.

**Comment by coreymckrill** [comment]
Commits that address the issues here are in #69.

**Comment by coreymckrill** [comment]
Resolved via https://meta.trac.wordpress.org/changeset/8713
